CONTRIBUTING.md

```diff
@@ -29,3 +29,142 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+to the specification.  Due to confusion in the past over Parquet versioning it
+is not expected that there will be a 3.x release of the specification in the
+foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backward compatible. A file written under an older version of the format
+   should be readable under a newer version of the format.
+
+2. Forward compatible. A file written under a newer version of the format with
+   the feature enabled can be read under an older version of the format, but
+   some information might be missing or performance might be suboptimal.
+
+3. Forward incompatible. A file written under a newer version of the format with
+   the feature enabled cannot be read under an older version of the format (e.g.
+   adding and using a new compression algorithm). It is expected any feature in
+   this category will provide a signal to older readers, so they can
+   unambiguously determine that they cannot properly read the file (e.g. via
+   adding a new value to an existing enum).
+
+New features are intended to be widely beneficial to users of Parquet, and
+therefore it is hoped third-party implementations will adopt them quickly after
+they are introduced. It is assumed that writing new parts of the format, and
+especially forward incompatible features, will be configured with a feature flag
+defaulted to "off", and at some future point the feature is turned on by default
+(reading of the new feature will typically be enabled without configuration or
+defaulted to on). Some amount of lead time is desirable to ensure a critical
+mass of Parquet implementations support a feature to avoid compatibility issues
+across the ecosystem.  Therefore, the Parquet PMC gives the following
+recommendations for managing features:
+
+1. Backward compatibility is the concern of implementations but given the
+   ubiquity of Parquet and the length of time it has been used, libraries should
+   support reading older versions of the format to the greatest extent possible.
+
+2. Forward compatible features/changes may be enabled and used by default in
+   implementations once the parquet-format containing those changes has been
+   formally released.  For features that may pose a significant performance
+   regression to older format readers, libaries should consider delaying default
+   enablement until 1 year after the release of the parquet-java implementation
+   that contains the feature implementation.
+
+3. Forward incompatible features/changes should not be turned on by default
+   until 2 years after the parquet-java implementation containing the feature is
+   released. It is recommended that changing the default value for a forward
+   incompatible feature flag should be clearly advertised to consumers (e.g. via
+   a major version release if using Semantic Versioning, or highlighed in
+   release notes).
```

Comment:
> We are talking of a Parquet V3, but are we thinking to increment a new major version more often for the purpose of making a feature that already exists on by default?

In parquet-java, no more then once per year.  But this is in essence of why we don't want to call anything parquet-v3.  At some point a new incompatible footer will be turned on.  When this is done for libraries that do semver like parquet java, the best way to get people to read the release notes is a major version bump.  I would also argue that according to semver changing the default for a forward incompatible feature really does constitute a major version bump.

Reply:
Discussed offline.  I'll add language to be more encouraging of people to turn things on themselves, rather than worrying too much about cadence of on by default.

Reply:
I added language discussing adoption below closer to parquet java.